sympy/polys/monomials.py

```diff
@@ -274,6 +274,49 @@ def monomial_gcd(A, B):
     """
     return tuple([ min(a, b) for a, b in zip(A, B) ])
 
+def monomial_ngcd(monomials):
+    """
+    Computes the greatest common divisor (GCD) of the exponents for each
+    variable in the monomials.
+
+    Parameters
+    ==========
+
+    monomials : iterable of tuples
+        Represents monomials, where each monomial is a tuple of exponents for variables.
+
+    Returns
+    =======
+
+    tuple
+        The greatest common divisor of the exponents for each variable in the
+        monomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.monomials import monomial_ngcd
+    >>> monomial_ngcd([(2, 3), (3, 2), (1, 4), (4, 1)])
+    (1, 1)
+
+    """
+    zero_monom = tuple(0 for _ in next(iter(monomials)))
+
+    # If the zero monomial is present in the monomials, return early
+    if zero_monom in monomials:
+        return zero_monom
+
+    gcd = next(iter(monomials))
+
+    for monom in monomials:
+        gcd = monomial_gcd(gcd, monom)
```

Comment:
How about
```
zero_monom = (0,) * len(monomials[0])
```
Why are `next` and `iter` being used here?

This loop will include the first item that was already extracted as `gcd`.